remove boilerplate + other little things

schollz opened the pull request:
made a couple of changes. feel free to reject any/all! also amenable to fixes. here they are:

1. moved boiler plate from most drones into the engine. now drones can simply be a file with the following:

```supercollider
{
    arg hz, amp;
    (SinOsc.ar(hz) * amp).dup;
}
```

the lagging is handled for each drone in Engine_Dronecaster now, and it opens up the possibility to add other boilerplate to the drones (like a limiter afterwards, or something) but mostly its probably a tiny bit easier to look at for a new person.

2. added Gristle 
3. added Dreamcrusher
4. sorted drones by name so the ordering is consistent
5. changed the controlspec parameters so that are integers when need be (the drones one was not...)
6. added a note parameter so you can change to specific note in parameters menu

tyleretters commented:
sorry i didn't see this PR sooner!! hahah oh man my first norns script. didn't even have tables in the lua...

---

@ryanlaws would you be willing to review the sc stuff, please? it is still over my head

ryanlaws commented on engine/Engine_Dronecaster.sc:
```diff
@@ -31,7 +31,13 @@ Dronecaster {
+			    var amp_, hz_;
+			    amp_ = Lag.ar(K2A.ar(amp), amplag);
+			    hz_ = Lag.ar(K2A.ar(hz), hzlag);
+			    drones[name].value(hz_,amp_);
```
Would these still work as named arguments? Kind of a nitpick but it might be a good future-proofing measure.

schollz replied:
do you mean, is it possible to do `drones[name].value(hz:hz_,amp:amp_)`? if so- ~~I don't the answer and will try and see.~~ it seems so:

```

f = { arg x =0,y=0; x.postln; y.postln };
f.value(y:2,x:1)
// 1
// 2
f.value(x:1,y:2)
// 1
// 2
```

ryanlaws commented on engine/drones/Gristle.scd:
```diff
@@ -0,0 +1,20 @@
+// a primal sawtooth
+
+{
+    arg hz=440, amp=0.02, amplag=0.02, hzlag=0.01;
```
Were these left in on purpose? Seems like the upstream `amplag` and `hzlag` won't get passed into this synth.

schollz replied:
thanks, thats a mistake, I can remove those.

ryanlaws commented on engine/drones/Sine.scd:
```diff
@@ -1,8 +1,5 @@
-    hz_ = Lag.ar(K2A.ar(hz), hzlag);
-    (SinOsc.ar(hz_) * amp_).dup;
+    arg hz, amp;
+    (SinOsc.ar(hz) * amp).dup;
```
So cute 👍🏻

ryanlaws commented:
Overall looks good, just a couple minor questions for you @schollz .

ryanlaws commented:
> sorry i didn't see this PR sooner!! hahah oh man my first norns script. didn't even have tables in the lua...
> 
> @ryanlaws would you be willing to review the sc stuff, please? it is still over my head

You got it, sorry for spacing.

schollz commented:
okay I think I fixed it up - I found a little error ("midi" in parameters should've been "note"). the named arguments work when you pass them to the function (is this what you meant @ryanlaws?) and I've updated to include that. 

let me know if there is anything else @tyleretters ! 

hope y'all enjoy dreamcrusher, its a treat.

ryanlaws commented:
Ahh good catch on `"midi"`. Yup, you got it on `hz:hz_` and `amp:amp_`!

Thank you for taking the time and effort to make these nice improvements! 😀
